File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.2/Python/list_count.py
def count(lista):
    if lista == []:
        return 0

    return 1 + count(lista[1:])

print(count([1, 3, 5, 7]))
print(count([2, 4, 6, 10, 12]))
print(count([3, 5, 7, 9, 13, 15]))

File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.2/Typescript/list_count.ts
function count(list: number[]): number {
  if (list.length === 0) {
    return 0;
  }

  return 1 + count(list.slice(1));
}

console.log(count([1, 3, 5, 7]));
console.log(count([2, 4, 6, 10, 12]));
console.log(count([3, 5, 7, 9, 13, 15]));
